File: algorithm/src/leetcode/ContainerWithMostWater_11.java
package leetcode;

// #11 - Container With Most Water

public class ContainerWithMostWater_11 {
	public static void main(String[] args) {
		int[] height = {1,8,6,2,5,4,8,3,7};

		System.out.println("maxArea = " + maxArea(height));
	}

	public static int maxArea(int[] height) {
		int s = 0;
		int e = height.length-1;

		int max = 0;
		while(s < e){
			int h = Math.min(height[s], height[e]);
			int width = (e-s) * h;
			max = Math.max(width, max);
			if(height[s] > height[e]){
				e--;
			}else{
				s++;
			}
		}

		return max;
	}
}

File: algorithm/src/leetcode/ruby/ContainerWithMostWater_11.rb
# 11 ruby Container With Most Water

# @param {Integer[]} height
# @return {Integer}
def max_area(height)
  s, e  = 0, height.length-1
  max = 0

  while(s < e)
    h = [height[s], height[e]].min
    w = (e-s)*h
    max = [w, max].max

    if(height[s] > height[e])
      e -= 1
    else
      s += 1
    end
  end

  max
end

height = [1,8,6,2,5,4,8,3,7]
p max_area(height)
